Automattic/wp-calypso
# Me: Cleanup legacy find friends and trophies config flags

#### Changes proposed in this Pull Request

* Remove always disabled `me/trophies` and `me/find-friends` config flag usage
* Unify redirect to `/me` by using the same controller method
* Remove unused duplicate controller methods
* Remove legacy unused config flags

#### Testing instructions

* Verify the removed config flags and controller methods are indeed not used anywhere.
* Verify `/me/trophies` redirect still works and redirects to `/me`.
* Verify `/me/find-friends` redirect still works and redirects to `/me`.
* Verify `/me` looks good.
* Verify all tests pass.
